test/stream/producer_daemon_test.go

```diff
@@ -115,7 +115,9 @@ func (s *ProducerDaemonTestSuite) TestWriteData() {
 		t: s.T(),
 	}
 
-	app := application.Default(application.WithLoggerHandlers(handler))
+	app := application.Default(application.WithLoggerHandlers(handler), application.WithKernelExitHandler(func(code int) {
+		assert.Equal(s.T(), kernel.ExitCodeOk, code, "exit code should be %d", kernel.ExitCodeOk)
+	}))
```

Comment:
```suggestion
	exitCodeHandler := func(code int) {
		assert.Equal(s.T(), kernel.ExitCodeOk, code, "exit code should be %d", kernel.ExitCodeOk)
	}
	handlers := []application.Option{
		application.WithLoggerHandlers(handler),
		application.WithKernelExitHandler(exitCodeHandler),
	}
	
	app := application.Default(handlers...)
```

Reply:
As this test is already in progress of being refactored, I will leave it like this..